src/login/signup.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import {
+  Text,
+  View,
+  StyleSheet,
+  TextInput,
+  TouchableHighlight,
+  Image
+} from 'react-native';
+import Hyperlink from 'react-native-hyperlink';
+
+export default class Signup extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {username: '', password: ''};
+  }
+
+  onPress() {
+    console.log(this.state.username);
+    this.props.navigator.push({
+      name: 'map'
+    })
+  }
+  
+  render() {
```

Comment:
Can any of this be refactored with the login page to make the code a bit more DRY?